consensus/fork.go

```diff
@@ -4,119 +4,115 @@ import (
 	"github.com/NebulousLabs/Sia/crypto"
 )
 
-// backtrackToBlockchain returns a list of nodes that go from the current node
-// to the first common parent. The common parent will be the final node in the
-// slice.
-func (s *State) backtrackToBlockchain(bn *blockNode) (nodes []*blockNode) {
-	nodes = append(nodes, bn)
+// invalidateNode recursively moves the children of a blockNode to the bad
+// blocks list.
+func (s *State) invalidateNode(node *blockNode) {
```

Comment:
Maybe this makes more sense somewhere else. I believe it's only ever called from diffs.go, might be worth putting in blocknode.go

Also, if it's not hard to implement, badBlocks should probably only contain the first parent, instead of recursively containing the parent and all the children.  All of the nodes need to be deleted, but only the first node needs to be added to badBlocks.

It's not a huge tragedy to put all of the nodes in badBlocks though.